pkg/controller/job/job_controller.go

```diff
@@ -475,23 +519,23 @@ func (jm *JobController) patchJobResource(j *batch.Job, pods []*v1.Pod) error {
 		resourceUpdates := mergeResourceChanges(pod, cMap)
 		if len(resourceUpdates) > 0 {
 			if requestVer, ok := pod.ObjectMeta.Annotations[schedulerapi.AnnotationResizeResourcesRequestVer]; ok && requestVer == j.ResourceVersion {
+
 				// This is not a new request, check if earlier request failed.
-				// TODO: Implement better scheme - exponential backoff/pod-leaving event.. For now update based retry - yuck.
-				skip := true
-				for i, podCondition := range pod.Status.Conditions {
+				for _, podCondition := range pod.Status.Conditions {
 					if podCondition.Type == v1.PodResourcesResizeStatus && podCondition.Status == v1.ConditionFalse {
-						if time.Since(podCondition.LastProbeTime.Time).Seconds() > 30 {
-							glog.Warningf("Resource resize request for pod %s by job %s version %s failed. Retrying after backoff..", pod.Name, j.Name, j.ResourceVersion)
-							pod.Status.Conditions[i].LastProbeTime = metav1.Now()
-							skip = false
+						// queue this job for resource update retry (when there's pod update and/or deletion)
+						if _, ok := jm.jobsToReSyncResource[j.UID]; !ok {
+
+							jm.jobsToReSyncResource[j.UID] = j
+							glog.V(4).Infof("Added job %s to retry resoruce queue", j.Name)
+
+							return nil
 						}
-						break
+
+						delete(jm.jobsToReSyncResource, j.UID)
+						glog.V(4).Infof("Retrying Resource resizing for pod %s by job %s version %s.", pod.Name, j.Name, j.ResourceVersion)
 					}
 				}
-				if skip {
-					glog.V(4).Infof("Resource resize was previously requested for pod %s by job %s version %s. Skipping request for now.", pod.Name, j.Name, j.ResourceVersion)
-					continue
-				}
 			}
 
 			anno := make(map[string]string)
```

Comment:
Do we need the 'pod.Annotations[schedulerapi.AnnotationResizeResourcesRequest] != anno[schedulerapi.AnnotationResizeResourcesRequest]' check below anymore considering we won't get here if a request is pending from the earlier check?

Reply:
removed